JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -18,44 +18,121 @@
 import java.sql.Statement;
 import java.sql.SQLTransientConnectionException;
 import java.sql.SQLInvalidAuthorizationSpecException;
+import java.util.Map;
+import java.util.HashMap;
 import java.util.Properties;
 import java.util.concurrent.Callable;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
 import picocli.CommandLine;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Option;
 import picocli.CommandLine.Parameters;
 
+import com.vertica.jdbc.VerticaConnection;
+
 @Command(name = "OAuthSampleApp", mixinStandardHelpOptions = true, version = "OAuth sample app 1.0", description = "Connects to a Vertica database using OAuth")
 public class OAuthSampleApp implements Callable<Integer> {
 
-    @Parameters(index = "0", description = "Host name")
-    private String host = "";
-
-    @Parameters(index = "1", description = "Database name")
-    private String dbName = "";
-
-    @Option(names = { "-p", "--port" }, description = "Port")
-    private String port = "5433";
+	private static Properties prop;
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+    private static String OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+   
+	private static Connection vConnection;
+	
+	private static String getConnectionString(String args) {
+		Map<String, String> kv=new HashMap<String, String>();
+        String[] entries = args.split(";");
 
-    @Option(names = { "-a", "--access-token" }, description = "Access token")
-    private String accessToken = "";
-
-    @Option(names = { "-r", "--refresh-token" }, description = "Refresh token")
-    private String refreshToken = "";
+        for (String entry : entries) {
+            String[] pair = entry.split("=");
+            kv.put(pair[0], pair[1]);
+        }
+        return "jdbc:vertica://" + kv.get("Host") + ":" 
```

Comment:
This gets the connection string from the command line. Do we want to get it from the properties file?

Reply:
This is not getting any values from command line. All the values we are getting from config.properties file. 
Parsed into Map and used here.